src/core/Form/Select/SingleSelect/SingleSelect.tsx

```diff
@@ -103,6 +104,20 @@ export interface InternalSingleSelectProps<T extends SingleSelectData> {
   forwardedRef?: React.RefObject<HTMLInputElement>;
 }
 
+type LoadingProps =
+  | {
+      loading?: false | never;
+      loadingText?: never;
```

Comment:
Could this still be a string instead of never? That way the text wouldn't have to be removed when setting loading to false, but it would still be required when loading is true? Same for MultiSelect.